lib/v2/screens/authentication/recover/recover_account_found/interactor/mappers/fetch_recover_guardian_state_mapper.dart

```diff
@@ -7,20 +9,56 @@ import 'package:seeds/v2/screens/authentication/recover/recover_account_found/in
 class FetchRecoverRecoveryStateMapper extends StateMapper {
   RecoverAccountFoundState mapResultToState(RecoverAccountFoundState currentState, RecoverGuardianInitialDTO result) {
     List<Result> members = result.membersData;
-    var linkResult = result.link;
+    Result linkResult = result.link;
+    Result userRecoversModel = result.userRecoversModel;
+    Result accountGuardians = result.accountGuardians;
 
     String? link;
     if (linkResult.isValue) {
       link = linkResult.asValue!.value;
     }
 
-    if (areAllResultsSuccess(members) && members.isNotEmpty && link != null) {
+    UserRecoversModel? userRecoversModelData;
+    if (userRecoversModel.isValue) {
+      userRecoversModelData = userRecoversModel.asValue!.value;
+    }
+
+    UserGuardiansModel? userGuardiansModel;
+    if (accountGuardians.isValue) {
+      userGuardiansModel = accountGuardians.asValue!.value;
+    }
+
+    if (areAllResultsSuccess(members) &&
+        members.isNotEmpty &&
+        link != null &&
+        userRecoversModelData != null &&
+        userGuardiansModel != null) {
```

Comment:
I think a comment above this` if condition` would be good to know when  is considered a success. With only the variables it is something complex for a dev not familiarized of this screen.

Reply:
agreed
added